Doc/c-api/object.rst

```diff
@@ -613,6 +613,36 @@ Object Protocol
 
    .. versionadded:: 3.14
 
+.. c:function:: int PyUnstable_Object_IsUniqueReferencedTemporary(PyObject *obj)
+
+   Check if *obj* is a unique temporary object
+   Returns ``1`` if *obj* is known to be a unique temporary object,
+   and ``0`` otherwise.  This function cannot fail, but the check is
+   conservative, and may return ``0`` in some cases even if *obj* is a unique
+   temporary object.
+
+   If an object is a unique temporary, it is guaranteed that the current code
+   has the only reference to the object. For arguments to C functions, this
+   should be used instead of checking if the reference count is ``1``. Starting
+   with Python 3.14, the interpreter internally avoids some reference count
+   modifications when loading objects onto the operands stack by
+   :term:`borrowing <borrowed reference>` references when possible, which means
+   that a reference count of ``1`` by itself does not guarantee that a function
+   argument uniquely referenced.
+
+   In the example below, ``my_func`` is called with a unique temporary object
+   as its argument::
+
+      my_func([1, 2, 3])
+
+   In the example below, ``my_func`` is **not** called with a unique temporary
+   object as its argument, even if its refcount is ``1``::
+
+      my_list = [1, 2, 3]
+      my_func(my_list)
+
```

Comment:
Can you add a link from Py_REFCNT to this function ("See also ... function")?

Py_REFCNT: https://docs.python.org/dev/c-api/refcounting.html#c.Py_REFCNT

Reply:
I suggest to add a link to PyUnstable_Object_IsUniqueReferencedTemporary() in Py_REFCNT() documentation.

Reply:
I added the link. I expect we'll rework the `Py_REFCNT` documentation to give better advice after we also add `PyUnstable_IsUniquelyReferenced`.